python/tank/platform/engine.py

```diff
@@ -70,6 +73,12 @@ def __init__(self, tk, context, engine_instance_name, env):
         # to access the invoker don't trip on undefined variables.
         self._invoker = None
         self._async_invoker = None
+
+        # By default we will assume that an engine does not support context
+        # changes. When changing contexts, the engine and all active apps
+        # will need to explicitly support the switch. If any do not, the
+        # engine will be torn down and restarted.
+        self._context_change_allowed = False
```

Comment:
Earlier, we brainstormed an approach where this was being less strict and the context switch would still happen even if all apps didn't implement the changing logic. Was that too risky/complicated? I am worried about having to release all our apps....